tests/check-refs.sh

```diff
@@ -40,3 +40,7 @@ nix-build -o $RESULT check-refs.nix -A test7
 
 # test10 should succeed (no disallowed references).
 nix-build -o $RESULT check-refs.nix -A test10
+
```

Comment:
Could you also add a test with a non-empty `referenceablePaths` (and making sure that a path in it will be properly considered)?